### PR TITLE
feat(split-ui): Slice 4 PR-A — prescribed-slot GET + on-plan history fields (DEC-089 D5)

### PR DESCRIPTION
**Pull request by leehopper**
**Slice 4 (Log & Log Book) PR-A — the backend root.** Ships the two pre-decided DEC-089 D5 wire deltas plus the slice's single codegen regen. Nothing consumes them yet; PR-B (`/log` banner) and PR-C (`/history` ledger) do.

Spec: `docs/specs/slice-4-log-logbook/spec.md` §3 PR-A (gitignored working-tree artifact).

## What's here

- **`GET /api/v1/workouts/logs/prescribed?date=` (D1)** — a thin, read-only wrapper over the *existing* `IWorkoutLogService.ResolveCandidatePrescriptionAsync`. No new resolution logic. Returns 200 + the prescription, or **200 + a literal `null` body** when the date resolves to none (off-plan / rest day / no active plan / malformed stored prescription / in-range week absent from `MicroWorkoutsByWeek`). Absence is an expected data state here, so this follows the `/conversation/turns` 200-on-absence precedent rather than `PlanRenderingController`'s 404, which is reserved for a genuinely missing primary resource. No antiforgery — it's a read, mirroring `POST /query` (DEC-055).
- **`PrescribedWorkoutDto` (Training module)** — a *deliberate* one-record sibling of Coaching's `CandidatePrescriptionDto`, not an accidental duplication. Module boundary favoured over cross-module reuse; the XML doc records why.
- **`WorkoutLogDto` gains `bool IsOnPlan` + `string? PrescribedWorkoutType` (D4)** — computed in `MapToDto` from the already-loaded `Prescription` complex property (table-split onto `WorkoutLog`'s own columns, always materialised, no `.Include()` needed). **Zero extra I/O.** Create path untouched — the create response never returns a `WorkoutLogDto`. Old/off-plan rows degrade to `false`/`null`.
- **Codegen** — `swagger.json` + RTK/zod clients regenerated, `PrescribedWorkoutDto` added to the hand-maintained `generated/index.ts` barrel (codegen doesn't touch that file; `tsc` is the gate, not `codegen:check`), and a hand-written `useGetPrescribedWorkoutQuery` wrapper in `workout-log.api.ts` following the `plan.api.ts`/`conversation.api.ts` precedent. `providesTags: ['Plan']` — the prescribed slot is plan-derived, and both `createWorkoutLog` and `regeneratePlan` already invalidate `Plan`.

## Two defects caught pre-merge

Neither was visible from reading the diff; both were caught by a failing test and by adversarial review.

1. **`Ok(null)` does not return 200 + null.** `ObjectResult`'s content-negotiation path selects the framework-registered `HttpNoContentOutputFormatter` for a null `Value` and silently rewrites the response to **204 No Content** — contradicting both the documented contract and the action's own `[ProducesResponseType(200)]`. The five null-branch integration tests failed on this. Fixed by returning `JsonResult`, which serializes `Value` directly and never enters formatter selection. A comment pins the reasoning so it doesn't get "simplified" back.
2. **An omitted `date` bound silently to `DateOnly.MinValue`.** `SimpleTypeModelBinder` treats an absent key as no-value and never touches `ModelState`, so `date` defaulted to `0001-01-01` → before any plan start → **200 + null**, i.e. a client bug masquerading as an ordinary off-plan day. Meanwhile a *malformed* `date` did 400, but that 400 was undeclared, so the generated clients promised a contract the API didn't honour. `date` is now `[BindRequired]` with the 400 declared as `ValidationProblemDetails` (matching `AuthController.Register`'s precedent for automatic-validation 400s, rather than this file's hand-thrown `Problem(...)` convention). Missing, empty and malformed now all fail honestly — three genuinely distinct ModelState branches, each pinned by a test. The generated arg tightens from `{ date?: string }` to `{ date: string }`.

## Decision record

`DEC-089` gains a one-line note ratifying the **D5(b) title source**: the field carries the **frozen `WorkoutType` enum name**, rendered client-side via the existing `WORKOUT_TYPE_LABELS`. Not the LLM-authored `WorkoutOutput.Title` — `RestructureDiffCalculator` diffs on `Title`, so live re-resolution would let a later adaptation restructure **retroactively rewrite the title on an already-logged ledger row**. `WorkoutPrescriptionSnapshot` has no title at all, and adding one would mean a migration for fidelity the design doesn't need. Point-in-time correct, zero migration, symmetric with the SSE card. This is a source clarification *within* the already-blessed D5(b), not a new wire delta — a full new DEC isn't warranted.

`PrescribedWorkoutType` is a plain `string?` **on purpose**: a nullable enum serializes as a `$ref` and trips the known `RequireNonNullablePropertiesSchemaFilter` bug (#163) that wrongly marks `$ref` props required. Verified in the regenerated swagger: `prescribedWorkoutType` is `nullable: true` and absent from `required`; `isOnPlan` is in `required`.

## Green

- Backend: `dotnet build` clean (0 warnings, TreatWarningsAsErrors), **2006/2006** tests.
- Frontend: **939/939** vitest, `npm run build` clean, `codegen:check` drift gate clean, eslint clean.
- **No LLM/prompt change → no eval re-record.**

Test coverage is net-new where it was absent: `MapToDto`'s field mapping was previously untested entirely (the existing service unit tests stub an empty repository list). The wire test asserts the raw camelCase JSON property names, so a serialization regression can't slip through a deserialize-only assertion.

## Notes for review

- The `ROADMAP.md` / `cycle-plan.md` edits are the previous session's "spec written" status, carried here rather than left dangling. Advancing the pointer to PR-B rides the usual post-merge docs PR.
- The rest-day and missing-micro-week tests deliberately exercise **different** null branches (day-lookup miss vs. week-lookup miss). The latter is a real production state today, not a hypothetical — per DEC-090, only week 1 of any plan currently gets micro workouts.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_011CFoccqV4SqYLLUAHjWvhc

## Summary by CodeRabbit

* **New Features**
  * Added support for retrieving the prescribed workout for a selected date.
  * Workout history now indicates whether an activity was part of the plan and displays its prescribed workout type when available.
  * Prescribed workout details include workout type, distance, duration, and pace ranges.

* **Documentation**
  * Updated Slice 4 roadmap, implementation planning, and design decision documentation.

* **Tests**
  * Added coverage for prescribed workout retrieval, validation, authentication, null results, and workout history mapping.

### CODE REVIEW

**Review by leehopper — COMMENTED** [review]
### Deep Review (Optimized mode)

No issues found. 7 dimensions reviewed (bugs, security, cross-file impact, test coverage, conventions & intent, type design, code simplification) across 21 changed files (3 high-risk, 3 medium-risk, 15 low-risk).

One type-design observation (WorkoutLogDto's IsOnPlan/PrescribedWorkoutType pairing is not structurally enforced) was raised, independently validated, and did not clear the confidence threshold for reporting — it is accurate but not currently reachable (single, correct construction call site).

This PR's own description already documents two pre-merge-caught defects (Ok(null) to JsonResult, missing [BindRequired] on date) that were independently verified as legitimate against actual ASP.NET Core content-negotiation and model-binding behavior.

---
Generated by deep-review | Reviewed up to: 90f40b1f7fd74594de8f6b72d206ecb4ce853c1c

**Comment by leehopper** [comment]
@coderabbitai full review

File: backend/src/RunCoach.Api/Modules/Training/Workouts/WorkoutLogsController.cs [hunk 1]
```diff
@@ -180,6 +181,46 @@ public async Task<IActionResult> QueryLogs(
...
+    /// </summary>
+    /// <param name="date">The date to resolve a prescription for.</param>
+    /// <param name="ct">Cancellation token.</param>
+    /// <returns>
+    /// 200 with the prescribed workout, or 200 with a <c>null</c> body when the date
+    /// resolves to no prescription (off-plan, rest day, no active plan, or a
+    /// malformed stored prescription) — the same absence-as-200 convention
+    /// <c>ConversationController.GetTurns</c> uses, never a 404; 400 when
+    /// <paramref name="date"/> is omitted or malformed ([ApiController]'s automatic
+    /// model-state validation, triggered by <see cref="BindRequiredAttribute"/>);
```
[the remark this answers is not in the document]
Comment (leehopper) [reply backend/src/RunCoach.Api/Modules/Training/Workouts/WorkoutLogsController.cs:198]: Fixed — removed the "Slice 4 D1" / "PR-A" planning-phase tags and the embedded `WorkoutLogService.cs` line reference from all four flagged locations, per the self-contained-comments convention. (f0f7df5)

File: backend/src/RunCoach.Api/Modules/Training/Workouts/WorkoutLogsController.cs [hunk 2]
```diff
@@ -180,6 +181,46 @@ public async Task<IActionResult> QueryLogs(
...
+
+        var snapshot = await service.ResolveCandidatePrescriptionAsync(userId, date, ct);
+
+        // JsonResult, not Ok(...): ObjectResult's content-negotiation path selects
+        // the framework-registered HttpNoContentOutputFormatter for a null Value,
+        // which silently rewrites this into a 204 with an empty body instead of
+        // the documented 200-with-literal-null contract. JsonResult serializes
+        // Value directly, bypassing that formatter-selection step entirely, so a
+        // null snapshot still round-trips as a literal JSON `null` at 200.
+        return new JsonResult(PrescribedWorkoutDto.FromSnapshot(snapshot));
```
[the remark this answers is not in the document]
Comment (leehopper) [reply backend/src/RunCoach.Api/Modules/Training/Workouts/WorkoutLogsController.cs:221]: Investigated, and the suggested fix doesn't actually work here — `typeof(PrescribedWorkoutDto?)` doesn't compile (CS8639: typeof cannot be used on a nullable reference type), and I also tried changing the action to `Task<ActionResult<PrescribedWorkoutDto?>>`, rebuilt, and inspected the regenerated swagger.json: the 200 response schema is still a bare `$ref` with no nullable marker either way.

Root cause: this repo's `SupportNonNullableReferenceTypes()` + `RequireNonNullablePropertiesSchemaFilter` (DEC-066/R-071) only infer nullability at the *property* level via NullabilityInfoContext on record properties — there's no existing mechanism that walks the action method's own return-type nullability onto the top-level response schema. Fixing this properly would need a new IOperationFilter, which is out of scope for this PR-A quick-nit pass. I've reverted the DTO/signature edit back to the original (working) `Task<IActionResult>` + `typeof(PrescribedWorkoutDto)`, and I'll track the operation-filter fix as a backlog item alongside the existing #163 nullable-schema tracking issue rather than ship a change that doesn't move the needle.
